frontend_tests/casper_tests/14-drafts.js

```diff
@@ -0,0 +1,208 @@
+var common = require('../casper_lib/common.js').common;
+
+function waitWhileDraftsVisible(then) {
+  casper.waitFor(function () {
+      return casper.evaluate(function () {
+          return $("#draft_overlay").length === 0 ||
+                 $("#draft_overlay").css("opacity") === "0";
+      });
+  }, then);
+}
+
+function waitUntilDraftsVisible(then) {
+  casper.waitFor(function () {
+      return casper.evaluate(function () {
+          return $("#draft_overlay").length === 1 &&
+                 $("#draft_overlay").css("opacity") === "1";
+      });
+  }, then);
+}
+
+common.start_and_log_in();
+
+casper.then(function () {
+    casper.test.info('Drafts page');
+
+    casper.waitUntilVisible('.drafts-link', function () {
+        casper.click('.drafts-link');
+        casper.then(function () {
```

Comment:
Generally we avoid nested `casper.then`; you should make the part from here down its own block.